controllers/kuadrant_controller.go

```diff
@@ -386,9 +393,19 @@ func (r *KuadrantReconciler) createOnlyInKuadrantNSCb(ctx context.Context, kObj
 			return err
 		}
 
+		switch obj := k8sObj.(type) {
+		case *appsv1.Deployment: // If it's a Deployment obj, it adds the required env vars
+			obj.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
+				{Name: "LIMITADOR_NAMESPACE", Value: kObj.Namespace, ValueFrom: &v1.EnvVarSource{}},
+				{Name: envLimitadorServiceName, Value: limitadorServiceName, ValueFrom: &v1.EnvVarSource{}},
+				{Name: envLimitadorServiceGrpcPort, Value: limitadorServiceGrpcPort, ValueFrom: &v1.EnvVarSource{}},
+			}
+			err = r.Client().Create(ctx, obj)
```

Comment:
not a big deal, but can you try to call create one time? For me it feels like objects are conditionally updated and then created. So something like this would work?

```go
var newObj client.Object                                                                    
// should not be copied, just another reference                                             
newObj = k8sObj                                                                             
                                                                                            
switch obj := k8sObj.(type) {                                                               
case *appsv1.Deployment: // If it's a Deployment obj, it adds the required env vars         
    obj.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{                                 
        {Name: "LIMITADOR_NAMESPACE", Value: kObj.Namespace, ValueFrom: &v1.EnvVarSource{}},
        {Name: envLimitadorServiceName, Value: limitadorServiceName, ValueFrom: &v1.EnvVarSource{}},
        {Name: envLimitadorServiceGrpcPort, Value: limitadorServiceGrpcPort, ValueFrom: &v1.EnvVarSource{}},
    }                                                                                       
    newObj = obj                                                                            
default:                                                                                    
}                                                                                           
                                                                                            
k8sObjKind := newObj.DeepCopyObject().GetObjectKind()                                       
err = r.Client().Create(ctx, newObj)                                                        
logger.V(1).Info("create resource", "GKV", k8sObjKind.GroupVersionKind(), "name", k8sObj.GetName(), "error", err)

```